### ISSUE TITLE
[P0][Phase-1][fix] Issue 1.5: Data Leakage in Wang Baseline Computation

### ISSUE THREAD

**Issue by matercomus** [matercomus/HOSER#5]
## Metadata
- **Priority**: P0-Critical
- **Phase**: Phase 1 (Week 1)
- **Feasibility**: fix
- **Category**: code-fix
- **Effort**: 6 hours
- **Dependencies**: None

## Problem Statement

Wang baseline statistics computed using full dataset including test set, causing data leakage. This inflates baseline performance unfairly.

## Evidence

**File**: Wang baseline computation scripts
- Statistics (mean, std) computed on entire dataset
- Test trajectories see training data statistics

## Required Changes

1. Modify Wang baseline to compute statistics only on training set
2. Apply training statistics to test set (no leakage)
3. Re-run Wang abnormality detection with clean baseline
4. Update results documentation

## Validation Steps
- [ ] Training/test split properly enforced
- [ ] Statistics computed only on training data
- [ ] Test set predictions use only training statistics
- [ ] Results re-computed and documented
- [ ] Performance comparison updated

## Files to Modify
- Wang baseline computation scripts
- `docs/WANG_ABNORMALITY_DETECTION_RESULTS.md`
- Evaluation scripts

---
**Reference**: `docs/implementation-review-remeditation/IMPLEMENTATION_REMEDIATION_REPORT.md` - Issue 1.5  
**Roadmap**: Phase 1 - Week 1 (6 hours)

**Comment by matercomus** [comment]
🤖 AI Agent started implementation

**Timestamp**: 2025-11-05T20:30:00 UTC

**Plan**:
1. Read detailed requirements from remediation report
2. Locate Wang baseline computation scripts
3. Identify where train/test pooling occurs
4. Modify to compute statistics only on training data
5. Ensure test set uses only training statistics (no leakage)
6. Re-run Wang abnormality detection
7. Update results documentation
8. Validate data leakage is eliminated

**Expected changes**:
- Fix Wang baseline computation to enforce train/test split
- Re-compute baselines and detection results
- Update documentation with corrected results
- Estimated effort: 6 hours

**Comment by matercomus** [comment]
## ⚠️ Reopening for Validation

**Status**: Code complete but NOT validated

The code fix has been implemented and committed, but the issue cannot be closed until:

1. ✅ Baseline files regenerated with train-only data
2. ⏳ Wang abnormality detection re-run
3. ⏳ Results compared (old vs new detection rates)
4. ⏳ Documentation updated if rates changed

**Baseline Regeneration Command**:
```bash
cd /home/matt/Dev/HOSER

# Backup old baselines
mv baselines baselines_old_with_leakage

# Regenerate with train-only data (~40 min total)
uv run python tools/compute_trajectory_baselines.py --dataset Beijing
uv run python tools/compute_trajectory_baselines.py --dataset BJUT_Beijing
uv run python tools/compute_trajectory_baselines.py --dataset porto_hoser

# Re-run Wang analysis
uv run python tools/analyze_wang_results.py
```

**Next Steps**:
- Run baseline regeneration overnight
- Compare detection rates
- Update `WANG_ABNORMALITY_DETECTION_RESULTS.md` if needed
- Close issue after validation complete
